pkg/sparrow/run.go

```diff
@@ -27,78 +27,75 @@ import (
 	"sync"
 	"time"
 
-	"github.com/caas-team/sparrow/pkg/factory"
-
-	"github.com/caas-team/sparrow/pkg/checks"
-
-	"github.com/caas-team/sparrow/pkg/checks/runtime"
-
-	"github.com/caas-team/sparrow/pkg/sparrow/targets"
-
 	"github.com/caas-team/sparrow/internal/logger"
 	"github.com/caas-team/sparrow/pkg/api"
+	"github.com/caas-team/sparrow/pkg/checks"
+	"github.com/caas-team/sparrow/pkg/checks/runtime"
 	"github.com/caas-team/sparrow/pkg/config"
 	"github.com/caas-team/sparrow/pkg/db"
-	"github.com/go-chi/chi/v5"
+	"github.com/caas-team/sparrow/pkg/factory"
+	"github.com/caas-team/sparrow/pkg/sparrow/targets"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
 const shutdownTimeout = time.Second * 90
 
 type Sparrow struct {
-	db db.DB
-	// the existing checks
-	checks map[string]checks.Check
-	server *http.Server
-
+	config  *config.Config
+	db      db.DB
+	api     api.API
+	loader  config.Loader
+	tarMan  targets.TargetManager
 	metrics Metrics
+	errorHandler
+	checkCoordinator
+}
 
+type checkCoordinator struct {
+	// the existing checks
+	checks      map[string]checks.Check
 	resultFanIn map[string]chan checks.Result
-
-	cfg *config.Config
-
 	// cRuntime is the channel where the loader sends the checkCfg configuration of the checks
 	cRuntime chan runtime.Config
 	// cResult is the channel where the checks send their results to
 	cResult chan checks.ResultDTO
+}
+
+type errorHandler struct {
```

Comment:
It really comforts me to have this (the sparrow struct was an eyesore for me since #5) and imo it makes the struct more readable but if you feel like this is unnecessary/complicating things I'll revert it.

Reply:
No need to revert, let's just roll with it. I do share your concern and thankfully the change is easy to revert or refactor.